sdk/core/Azure.Core.Amqp/src/AmqpMessageId.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Core.Amqp
+{
+    /// <summary>
+    /// Represents the AMQP message ID.
+    /// <seealso href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-message-id-ulong"/>.
+    /// </summary>
+    public struct AmqpMessageId : IEquatable<AmqpMessageId>
+    {
+        private readonly string? _messageIdString;
+
+        /// <summary>
+        /// Initializes a new <see cref="AmqpMessageId"/> using the provided
+        /// <see cref="string"/>.
+        /// </summary>
+        ///
+        /// <param name="messageId">The message Id.</param>
+        public AmqpMessageId(string messageId)
```

Comment:
Should we override the equality and inequality operators because we're overriding equals and implementing IEquatable<T>?

Reply:
Same as with Address - the additional formats would require an allocation.